### PR TITLE
POST method to save paper lists

### PR DESCRIPTION
**Pull request by Bera0422**
I added the POST method implementation which enables users to save a paper list. I also wrote the relevant unit tests. Also the url is added to urls.py

### CODE REVIEW

File: practice_app/api/views.py
Comment (laylaylo) [inline practice_app/api/views.py:452]: Are you using these username and password variables anywhere? Please delete these lines, if they are redundant.

File: practice_app/api/tests.py
Comment (laylaylo) [inline practice_app/api/tests.py:470]: Could you please add a test case for empty and null paper list IDs to ensure that these cases are being handled in implementation? I doubt that you're not handling these cases in save_paper_list method.

**Review by laylaylo — CHANGES_REQUESTED** [review]
Thank you for your work! Please consider my comments and merge conflicts from the main in your PR. After the changes made by @hakanaktas0 merged into your PR, all checks done by Django CI should be passed.

File: practice_app/api/views.py
Comment (Bera0422) [reply practice_app/api/views.py:452]: I used them in the next line to authenticate the user.

Comment (laylaylo) [reply practice_app/api/views.py:452]: I believe you're talking about the line 435 and 436

Comment (Bera0422) [reply practice_app/api/views.py]: Oh, I noticed now. I removed those redundant lines.

**Review by laylaylo — APPROVED** [review]
All good! It can be merged after passing checks of workflow.

**Review by uyduranomer — APPROVED** [review]
Very good implementation. I think this pull request is ready to be merged.

**Pull request merged**
